cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1045 @@
+
+    p_ann_pruned = total_ann and num_ann_to_zero / total_ann or 0  # avoid division by zero
+    if p_ann_pruned > MAX_ALLOWED_BOX_PRUNE:
+        warnings.warn(
+            f"Pruning with threshold=={threshold} prunes {p_ann_pruned}% labels. Consider lowering the threshold.",
+            UserWarning,
+        )
+    if verbose:
+        print(f"Pruning {num_ann_to_zero} labels out of {total_ann} using threshold=={threshold}.")
+    return predictions_copy
+
+
+def _assert_valid_inputs(
+    labels: List[Dict[str, Any]],
+    predictions,
+    method: Optional[str] = None,
+    threshold: Optional[float] = None,
+):
+    """Asserts proper input format."""
+    if len(labels) != len(predictions):
+        raise ValueError(
+            f"labels and predictions length needs to match. len(labels) == {len(labels)} while len(predictions) == {len(predictions)}."
+        )
+    # Typecheck labels and predictions
+    if not isinstance(labels[0], dict):
+        raise ValueError(
+            f"Labels has to be a list of dicts. Instead it is list of {type(labels[0])}."
+        )
+    # check last column of predictions is probabilities ( < 1.)?
+    if not isinstance(predictions[0], np.ndarray):
+        raise ValueError(
+            f"Prediction has to be a list of np.ndarray. Instead it is list of {type(predictions[0])}."
+        )
+    if not predictions[0][0].shape[1] == 5:
+        raise ValueError(f"Prediction values have to be of format [_,_,_,_,pred_prob].")
+
+    valid_methods = ["objectlab"]
+    if method is not None and method not in valid_methods:
+        raise ValueError(
+            f"""
+            {method} is not a valid object detection scoring method!
+            Please choose a valid scoring_method: {valid_methods}
+            """
+        )
+
+    if threshold is not None and threshold > 1.0:
+        raise ValueError(
+            f"""
+            Threshold is a cutoff of predicted probabilities and therefore should be <= 1.
+            """
+        )
+
+
+def visualize(
+    image_path: str,
+    label: Dict[str, Any],
+    prediction: np.ndarray,
+    *,
+    prediction_threshold: Optional[float] = None,
+    given_label_overlay: bool = True,
+    class_labels: Optional[Dict[Any, Any]] = None,
+    figsize: Optional[Tuple[int, int]] = None,
+):
+    """Visualize bounding box labels (given labels) and model predictions for an image. The given labels
+    are shown with red while the predictions are shown in blue.
+
+    A single image is visualized with ``L`` annotated bounding boxes, ``M`` predicted bounding boxes and ``K`` total classes.
+
+
+    Parameters
+    ----------
+    image_path:
+        Full path to the image file.
+
+    label:
+        The given label for a single image in the format ``{'bboxes': np.ndarray((L,4)), 'labels': np.ndarray((L,))}`` where
+        ``L`` is the number of bounding boxes for the `i`-th image and ``bboxes[j]`` is in the format ``[x,y,x,y]`` with given label ``labels[j]``.
```

Comment:
Could we move all code documentation to one function? 
and refer to them here?
Example: 

https://github.com/cleanlab/cleanlab/blob/e33b757fa0fbf01288c84e341636c79f68588cb4/cleanlab/multilabel_classification/filter.py#L148

We add a one sentence description and ask user to refer to document under an original definition, this will avoid repetition, and in case we make doc changes, we only do it once